api/src/main/resources/workbench.yaml

```diff
@@ -57,6 +57,13 @@ parameters:
     type: integer
     format: int64
     required: true
+  conceptSetId:
```

Comment:
I actually want to follow the same pattern as cohorts; I think there are good reasons for it.

Concept set names (like cohort names) are not URL safe, and are mutable, so having them in the path probably doesn't make sense. (I want to be able to say 'rename the concept set with ID X to "Blah blah blah"'... this allows us to do that.)

You *could* refer to cohorts or concept set IDs by ID instead of by name in the cohort materialization API, but we don't expect researchers to actually know the internal ID of their cohort or concept set... so they would have to look it up programatically by name... at which you point you'd want to give them a convenience method that didn't require them to do that... and then you could save yourself a trip to the server by just accepting the cohort name or concept set name instead of the ID. So that's why we use IDs there.

Furthermore, when you clone a workspace, the cohort and concept set IDs will change, but the names of those things won't. If you want existing notebooks to keep working, they need to look them up by name, not ID.

So I think we have good reasons to use an immutable ID for the PK and REST path, but names when referencing cohorts and concept sets in notebooks. Let me know if this sounds reasonable to you.

Reply:
BTW, another option we considered early on (but decided against) was having an immutable human readable ID based off the name, scoped to the workspace. That could be used in both the REST path and notebooks (and would avoid issues where a cohort rename would break notebooks). But then you have a problem where when you change the name, the immutable ID no longer matches, and you can't change it. We decided that it was simpler and better to be able to change names and just warn users that doing so may break notebooks.